PythonVisualizations/SimpleQueue.py

```diff
@@ -6,153 +6,119 @@
-            #deal with wraparound
-            if self.front == 0:
-                self.front = self.size
-            
-            #decrement front
-            self.front -= 1
-                        
-            # create new cell and cell value display objects
-            # Start drawing new one at rear
-            cell = canvas.create_rectangle(ARRAY_X0+CELL_SIZE*self.front, ARRAY_Y0, \
-                                           ARRAY_X0+CELL_SIZE*(self.front+1), ARRAY_Y0 + CELL_SIZE, \
-                                           fill=Queue.colors[Queue.nextColor], outline='')
-            cell_val = canvas.create_text(ARRAY_X0+CELL_SIZE*self.front + (CELL_SIZE / 2), \
-                                          ARRAY_Y0 + (CELL_SIZE / 2), text=val, font=('Helvetica', '20'))
-            
-            #insert the item
-            self.list[self.front] = drawable(val, Queue.colors[Queue.nextColor], cell, cell_val)
-            
-            self.nItems += 1
-            
-            # increment nextColor
-            Queue.nextColor = (Queue.nextColor + 1) % len(Queue.colors)
-            
-            self.onOffButtons()
-            
-            # update window
-            window.update()
+        color = drawable.palette[self.nextColor]
+        # increment nextColor
+        self.nextColor = (self.nextColor + 1) % len(drawable.palette)
+        # deal with wraparound
+        if self.rear == self.size - 1:  # deal with wraparound
+            self.rear = -1
+            self.moveItemsBy(self.rearArrow, (self.CELL_SIZE * self.size * -1, 0))
+        #move arrow
+        self.moveItemsBy(self.rearArrow, (self.CELL_SIZE, 0))
+        # increment rear
+        self.rear += 1
+
+        # create new cell and cell value display objects
+        # Start drawing new one at rear
+        cell = self.canvas.create_rectangle(self.ARRAY_X0+self.CELL_SIZE*self.rear, self.ARRAY_Y0, \
+                                       self.ARRAY_X0+self.CELL_SIZE*(self.rear+1), self.ARRAY_Y0 + self.CELL_SIZE, \
+                                       fill=color, outline='')
+        cell_val = self.canvas.create_text(self.ARRAY_X0+self.CELL_SIZE*self.rear + (self.CELL_SIZE / 2), \
+                                      self.ARRAY_Y0 + (self.CELL_SIZE / 2), text=val, font=('Helvetica', '20'))
+
+       # insert the item
+        self.list[self.rear] = drawable(val, color, cell, cell_val)
+        #increment number of items
+        self.nItems += 1
 
+        self.onOffButtons()
+        # update window
+        self.window.update()
+        self.cleanUp(callEnviron)
+        self.stopAnimations()
 
-    
     # remove the left element of the queue, or None if empty
     def removeFront(self):
-        
+        callEnviron = self.createCallEnvironment()
+        self.startAnimations()
         #if the queue is empty, exit
-        if self.size == 0:
+        if self.nItems == 0:
+            self.setMessage('Queue is empty!')
             return None
```

Comment:
if the delete button is disabled if there are no elements should i get rid of this error message, or keep it?

Reply:
If the disabling works properly, the method could never be called when the queue is empty.  That said, it's nice to have backup in case the button somehow is enabled.